clang/lib/CodeGen/Targets/AMDGPU.cpp

```diff
@@ -105,6 +105,11 @@ void AMDGPUABIInfo::computeInfo(CGFunctionInfo &FI) const {
   if (!getCXXABI().classifyReturnType(FI))
     FI.getReturnInfo() = classifyReturnType(FI.getReturnType());
 
+  // srets / indirect returns are unconditionally in the alloca AS.
+  if (FI.getReturnInfo().isIndirect())
+    FI.getReturnInfo().setIndirectAddrSpace(
+        getDataLayout().getAllocaAddrSpace());
```

Comment:
It's also ok to just hardcode to AMDGPUAS::PRIVATE_ADDRESS

Reply:
> I'm assuming these should default to DL.getAllocaAddrSpace but I guess that's a separate change that calls for new test coverage

I think it's just a slightly different path for what this patch is already doing: IRGen is currently assuming that `sret` is always in addrspace 0, and the patch is fixing that. Feels like we should cover the C++ case at the same time unless there's a significant reason not to.

Reply:
@rjmccall are you OK with the current solution (@arsenm  appears to be)?